vmdb/app/models/mixins/authentication_mixin.rb

```diff
@@ -74,8 +91,19 @@ def update_authentication(data, options = {})
     data.each_pair do |type, value|
       cred = self.authentication_type(type)
       current = {:new => nil, :old => nil}
-      current[:new] = {:user => value[:userid], :password => value[:password]} unless value[:userid].blank?
-      current[:old] = {:user => cred.userid, :password => cred.password} if cred
+
+      if value[:auth_key]
+        cred = self.auth_user_keypair(type)
+        # TODO(lsmola) figure out if there is a better way. Password field is replacing \n with \s, I need to replace
+        # them back
+        fixed_auth_key = value[:auth_key].gsub(/-----BEGIN\sRSA\sPRIVATE\sKEY-----/, '')
+        fixed_auth_key = fixed_auth_key.gsub(/-----END\sRSA\sPRIVATE\sKEY-----/, '')
+        fixed_auth_key = fixed_auth_key.gsub(/\s/, "\n")
+        value[:auth_key] = '-----BEGIN RSA PRIVATE KEY-----' + fixed_auth_key + '-----END RSA PRIVATE KEY-----'
```

Comment:
@martinpovolny I think textarea should be fine. It's the text input (password field) that replaces \n to space. I think it even happens on browser side, when you copy paste it.

Alright, I'll check the file upload, that might be the easiest option here. Maybe I'll send it as followup PR. Maybe allowing both of the options. Unless you feel like puking from ^ solution, then I will just replace it with file upload. :-D But I think I might do that anyway